core/new-gui/src/app/workspace/component/navigation/navigation.component.ts

```diff
@@ -371,4 +391,36 @@ export class NavigationComponent implements OnInit {
       });
   }
 
+  handleDisableOperatorIconChange() {
+    Observable.merge(
+      this.workflowActionService.getJointGraphWrapper().getJointOperatorHighlightStream(),
+      this.workflowActionService.getJointGraphWrapper().getJointOperatorUnhighlightStream(),
+      this.workflowActionService.getJointGraphWrapper().getJointGroupHighlightStream(),
+      this.workflowActionService.getJointGraphWrapper().getJointGroupUnhighlightStream(),
+      this.workflowActionService.getTexeraGraph().getDisabledOperatorsChangedStream(),
+    ).subscribe(event => {
+      this.isDisableOperator = this.calculateIsDisableOperator();
+      this.isDisableOperatorClickable = this.effectivelyHighlightedOperators().length !== 0;
+    });
+  }
+
+  calculateIsDisableOperator(): boolean {
+    const allDisabled = this.effectivelyHighlightedOperators().every(
+      op => this.workflowActionService.getTexeraGraph().isOperatorDisabled(op));
```

Comment:
if the disabled property is on the operator itself, isn't that easier to check if it is disabled or not?

Reply:
I don't see a difference here because the api will be the same, the highlighted operator is an operator ID instead of the operator object